### ISSUE TITLE
Infisical Plugin

### ISSUE THREAD

**Issue by DanielHougaard** [jenkins-infra/repository-permissions-updater#3924]
### Repository URL

https://github.com/Infisical/infisical-jenkins

### New Repository Name

infisical-plugin

### Description

The Infisical Plugin is a native integration that links Infisical with Jenkins, without the need to use our CLI tool as a part of the freestyle Jenkins project.

The plugin pulls in secret values from Infisical and sets environment variables that can be used in the build step(s). The secret values are masked from the output.

The plugin takes use of Credentials to securely store the Infisical authentication credentials.

### GitHub users to have commit permission

@danielhougaard
@maidul98
@dangtony98
@vmatsiiako
@akhilmhdh

### Jenkins project users to have release permission

danielhougaard

### Issue tracker

Jira

**Comment by jenkins-cert-app** [comment]
<b>Security audit, information and commands</b>

The security team is auditing all the hosting requests, to ensure a better security by default.

This message informs you that a [Jenkins Security Scan](https://www.jenkins.io/redirect/jenkins-security-scan/) was triggered on your repository.
It takes ~10 minutes to complete.

<details><summary>Commands</summary>

The bot will parse all comments, and it will check if any line start with a command.

Security team only:
<ul>
	<li><code>/audit-ok</code> => the audit is complete, the hosting can continue :tada:.</li>
	<li><code>/audit-skip</code> => the audit is not necessary, the hosting can continue :tada:.</li>
	<li><code>/audit-findings</code> => the audit reveals some issues that require corrections :pencil2:.</li>
</ul>

Anyone:
<ul>
	<li><code>/request-security-scan</code> => the findings from the <a href="https://www.jenkins.io/redirect/jenkins-security-scan/" rel="nofollow">Jenkins Security Scan</a> were corrected, this command will re-scan your repository :mag:.</li>
	<li><code>/audit-review</code> => the findings from the audit were corrected, this command will ping the security team to review the findings :eyes:.
	It's only applicable when the previous audit required changes.</li>
</ul>

<i>Only one command can be requested per comment.</i>

</details>

<i>(automatically generated message, version: 1.28.4)</i>

**Comment by jenkins-cert-app** [comment]
The Jenkins Security Scan discovered 5 finding(s) :mag:.

Please follow the instructions below for every identified issues:
- Implement the recommended fix to address the issue.
- If you think it's a false positive, suppress the warning directly within the code.
- Alternative, you write an explanation here about why you think it's irrelevant. That will require a manual review, leading to a slower process.

After addressing the findings through one of the above methods:
- If all modifications have been made to the code, please initiate a new security scan by triggering the `/request-security-scan` command.
- If there are any unresolved findings (those not corrected or suppressed), request a review from the Jenkins security team by using the `/audit-review` command.

---
## Jenkins: Missing permission check on a form fill web method with credentials lookup

You can find detailed information about this finding [here](https://github.com/jenkins-infra/jenkins-codeql/blob/main/src/CredentialsEnumeration.md).

<details><summary><a href="https://github.com/Infisical/infisical-jenkins/blob/6e9605ba7ab3ddae72b615c15b1ebcce142542be/src/main/java/io/jenkins/plugins/infisicaljenkins/configuration/InfisicalConfiguration.java#L140">InfisicalConfiguration.java#140</a></summary>

```
doFillInfisicalCredentialIdItems should perform a permission check before calling #includeAs
doFillInfisicalCredentialIdItems should perform a permission check before calling #includeEmptyValue
```

</details>



## Stapler: Missing POST/RequirePOST annotation

You can find detailed information about this finding [here](https://github.com/jenkins-infra/jenkins-codeql/blob/main/src/WebMethodMissingPostAnnotation.md).

<details><summary><a href="https://github.com/Infisical/infisical-jenkins/blob/6e9605ba7ab3ddae72b615c15b1ebcce142542be/src/main/java/io/jenkins/plugins/infisicaljenkins/configuration/InfisicalConfiguration.java#L140">InfisicalConfiguration.java#140</a></summary>

```
Potential CSRF vulnerability: If DescriptorImpl#doFillInfisicalCredentialIdItems connects to user-specified URLs, modifies state, or is expensive to run, it should be annotated with @POST or @RequirePOST
```

</details>



## Stapler: Missing permission check

You can find detailed information about this finding [here](https://github.com/jenkins-infra/jenkins-codeql/blob/main/src/WebMethodMissingPermissionCheck.md).

<details><summary><a href="https://github.com/Infisical/infisical-jenkins/blob/6e9605ba7ab3ddae72b615c15b1ebcce142542be/src/main/java/io/jenkins/plugins/infisicaljenkins/configuration/InfisicalConfiguration.java#L140">InfisicalConfiguration.java#140</a></summary>

```
Potential missing permission check in DescriptorImpl#doFillInfisicalCredentialIdItems
```

</details>



## Jenkins: Plaintext password storage

You can find detailed information about this finding [here](https://github.com/jenkins-infra/jenkins-codeql/blob/main/src/PlaintextPasswordStorage.md).

<details><summary><a href="https://github.com/Infisical/infisical-jenkins/blob/6e9605ba7ab3ddae72b615c15b1ebcce142542be/src/main/java/io/jenkins/plugins/infisicaljenkins/model/InfisicalSecretValue.java#L17">InfisicalSecretValue.java#17</a></summary>

```
Field should be reviewed whether it stores a password and is serialized to disk: infisicalKey
```

</details>


<details><summary><a href="https://github.com/Infisical/infisical-jenkins/blob/6e9605ba7ab3ddae72b615c15b1ebcce142542be/src/main/java/io/jenkins/plugins/infisicaljenkins/model/SingleSecretResponse.java#L7">SingleSecretResponse.java#7</a></summary>

```
Field should be reviewed whether it stores a password and is serialized to disk: secretKey
```

</details>

**Comment by jenkins-cert-app** [comment]
The Jenkins Security Scan did not find anything dangerous with your plugin, congratulations! :tada:

---

:bulb: The Security team recommends that you are setting up the scan in your repository by following [our guide](https://www.jenkins.io/redirect/jenkins-security-scan/).

**Comment by DanielHougaard** [comment]
Docs (README) has been added to the repository.

**Comment by mawinter69** [comment]
> * Docs (README) has been added to the repository.
> * A separate plugin for pipeline support will be raised once this plugin is released, as it will use this plugin as a dependency.
>   
>   * I've seen examples of other plugins doing this as well.

Why a separate plugin? It is much more common to have the things for freestyle and pipeline in the same plugin. And wouldn't it be enough to give your buildwrapper and InfisicalSecret  a `@Symbol` annotation on the Descriptor so you can use it in pipeline (just guessing here, never tried this myself for a buildwrapper)?

**Comment by DanielHougaard** [comment]
Hi @mawinter69, thanks for the clarification. After taking a second look at the docs I totally agree that it would be senseless to create a separate plugin for pipeline support. I've updated the repo to support pipelines!

**Comment by DanielHougaard** [comment]
@mawinter69 Just following up, did you get to take a look after my update from 2 days ago?

**Comment by mawinter69** [comment]
one comment, maybe change the symbol from `infisical` to `withInfisical`, similar to e.g. `withCredentials` 

Also if I need to configure infisical in each pipeline, that will cause a lot of duplicate work. Isn't it likely that almost all pipelines will use the same configuration and just need different secrets? So it might be good to provide one or more central configurations that can be reused.

And as it looks as if infisical is providing credentials, I wonder if there is a way to say something like
```
withCredentials([infisicalSecret(config: configID, path: '/', secretValues: [...]
)]) {
...
}
```

Just proposal I'm not using infisical (and most likely will never use it)

**Comment by DanielHougaard** [comment]
Hi @mawinter69, I've updated the `infisical` symbol to `withInfisical`. 

In regards to the config, it's likely that the environment slug and project slug are unique on a per-job basis.
We may release an update in the future, allowing for a more "global" config. For now we're looking to release with the current config implementation.

**Comment by DanielHougaard** [comment]
Hi again @mawinter69, we're looking to get this done as soon as possible (ideally Monday in the coming week). If you could please let us know how we can proceed that would be great!

**Comment by mawinter69** [comment]
I think you have everything to release for now. My suggestions can also be delivered later.

**Comment by DanielHougaard** [comment]
@mawinter69 Awesome! Please let me know when everything is set up, and I’ll get started with the release:-)

**Comment by NotMyFault** [comment]
Hey,

I took a brief look over your hosting request and have some feedback for you:
- https://github.com/Infisical/infisical-jenkins/blob/6709df615e40bb23c9667d78c6424b6d807a277a/pom.xml#L33 for CD, this should be 999999-SNAPSHOT
- https://github.com/Infisical/infisical-jenkins/blob/6709df615e40bb23c9667d78c6424b6d807a277a/pom.xml#L36 Following the sample for new plugins, this should be set to false.
- Instead of shading json yourself, you can depend on the corresponding Jenkins plugin: https://plugins.jenkins.io/json-api/dependencies/
- https://github.com/Infisical/infisical-jenkins/blob/6709df615e40bb23c9667d78c6424b6d807a277a/pom.xml#L57-L61 same like above: https://plugins.jenkins.io/gson-api/dependencies/
- https://github.com/Infisical/infisical-jenkins/blob/6709df615e40bb23c9667d78c6424b6d807a277a/pom.xml#L62-L101 Remove all plugins your plugin doesn't necessarily depend on.
- https://github.com/Infisical/infisical-jenkins/blob/6709df615e40bb23c9667d78c6424b6d807a277a/pom.xml#L44 should always match your jenkins.version specified.

**Comment by DanielHougaard** [comment]
Hi @NotMyFault, thank you for the feedback! I've implemented all your feedback in the repo.

I have one question/need for clarification:

> https://github.com/Infisical/infisical-jenkins/blob/6709df615e40bb23c9667d78c6424b6d807a277a/pom.xml#L44 should always match your jenkins.version specified.

https://github.com/Infisical/infisical-jenkins/commit/de68abd6e4fc7f0898fb8c7a987371f8ac8d1d67

In the commit above I've updated the version as requested. Does it need to be an ***exact*** match? Or is it fine to end it with `.x` like I have in the commit above? Thanks!
